### PR TITLE
Enhancements for test case OCP-18082

### PR DESCRIPTION
**Pull request by lenahorsley**
Polarion test case: OCP-18082 - NextGen Pod scheduler at capacity with Node Affinity and Anti-Affinity rules	https://polarion.engineering.redhat.com/polarion/#/project/OSE/workitem?id=OCP-18082

Jenkins job: https://mastern-jenkins-csb-openshift-qe.apps.ocp-c1.prod.psi.redhat.com/job/scale-ci/job/e2e-benchmarking-multibranch-pipeline/job/regression-test/38/console

### CODE REVIEW

**Comment by lenahorsley** [comment]
@paigerube14 
@qiliRedHat

File: perfscale_regression_ci/scripts/scalability/node-affinity-anti-affinity.sh [hunk 1]
```diff
@@ -0,0 +1,171 @@
...
+source node-affinity-anti-affinity_env.sh
+
+function show_node_labels() {
+  oc get node --show-labels
+  oc get node -l cpu=4
+  oc get node -l cpu=6
+  oc get node -l beta.kubernetes.io/arch=intel
+}
+
+function check_no_error_pods()
```
Comment (paigerube14) [inline perfscale_regression_ci/scripts/scalability/node-affinity-anti-affinity.sh:24]: Let's move this to the common functions in the scripts folder

File: perfscale_regression_ci/scripts/scalability/node-affinity-anti-affinity.sh [hunk 2]
```diff
@@ -0,0 +1,171 @@
...
+  error=`oc get pods -n $1 | grep Error | wc -l`
+  if [ $error -ne 0 ]; then
+    echo "$error pods found, exiting"
+    #loop to find logs of error pods?
+    exit 1
+  fi
+}
+
+export ANTI_AFFINITY_JOB_ITERATION=${ANTI_AFFINITY_JOB_ITERATION:-130}
+export AFFINITY_JOB_ITERATION=${AFFINITY_JOB_ITERATION:-200}
```
Comment (paigerube14) [inline perfscale_regression_ci/scripts/scalability/node-affinity-anti-affinity.sh:35]: I think we should be able to run this test with both of the job iterations here set to 190. That's what I've run in previous release regression runs

File: perfscale_regression_ci/scripts/scalability/node-affinity-anti-affinity.sh [hunk 3]
```diff
@@ -0,0 +1,171 @@
...
+show_node_labels
+
+sleep 5
+
+
+echo "======Use kube-burner to load the cluster with test objects======"
+run_workload
+
+
+echo "======Waiting for the pods in both namespaces======"
```
Comment (paigerube14) [inline perfscale_regression_ci/scripts/scalability/node-affinity-anti-affinity.sh:99]: I think this should be covered in the run_workload. I think it should wait for all the pods to go running before showing success. 

We might need to add in a check of pass/fail after run_workload, think we can do this in separate PR if you don't want to tackle

File: perfscale_regression_ci/scripts/scalability/node-affinity-anti-affinity.sh [hunk 4]
```diff
@@ -0,0 +1,171 @@
...
+echo "node_anti_affinity_pods_expected $node_anti_affinity_pods_expected"
+
+if [ $node_anti_affinity_pods_expected == $node_anti_affinity_pods_actual ]; then
+  echo -e "Actual $node_anti_affinity_pods_actual pods were sucessfully deployed.  Node Anti-affinity test passed!"
+  (( ++pass_or_fail ))
+else
+  echo -e "Actual $node_anti_affinity_pods_actual pods deployed does NOT match expected $node_anti_affinity_pods_expected pods for node Anti-affinity test. Node Anti-affinity test failed !"
+fi
+
+sleep 60
```
Comment (paigerube14) [inline perfscale_regression_ci/scripts/scalability/node-affinity-anti-affinity.sh:144]: Think this is unnecessary sleep here we can get rid of

File: perfscale_regression_ci/scripts/scalability/node-affinity-anti-affinity.sh [hunk 5]
```diff
@@ -0,0 +1,171 @@
...
+if [[ ${pass_or_fail} == 2 ]]; then
+  echo -e "\nOverall Node Affinity and Anti-affinity Testcase result:  PASS"
+  echo "======Clean up test environment======"
+  # # delete projects:
+  # ######### Clean up: delete projects and wait until all projects and pods are gone
+  echo "Deleting test objects"
+  delete_project_by_label kube-burner-job=$AFFINTIY_NAME
+  delete_project_by_label kube-burner-job=$ANTI_AFFINTIY_NAME
+
+  sleep 30
```
Comment (paigerube14) [inline perfscale_regression_ci/scripts/scalability/node-affinity-anti-affinity.sh:157]: don't think we need this sleep here, the delete_project_by_label will take care of the waiting till the project is gone

File: perfscale_regression_ci/kubeburner-object-templates/node-affinity-anti-affinity-config.yml [hunk 6]
```diff
@@ -0,0 +1,41 @@
...
+    burst: ${BURST}
+    namespacedIterations: false
+    podWait: false
+    verifyObjects: true
+    objects:
+    - objectTemplate: "hello_node-anti-affinity.yml"
+      replicas: 1
+    - objectTemplate: "node-affinity-service.yml"
+      replicas: 1
+      inputVars:
```
Comment (paigerube14) [inline perfscale_regression_ci/kubeburner-object-templates/node-affinity-anti-affinity-config.yml:35]: I don't we need a service here, just the anti-affinity.yml  and node-affinity.yml

File: perfscale_regression_ci/scripts/scalability/node-affinity-anti-affinity.sh (hunk 2 above)
Comment (lenahorsley) [reply perfscale_regression_ci/scripts/scalability/node-affinity-anti-affinity.sh]: done.

**Comment by lenahorsley** [comment]
@paigerube14 @qiliRedHat 

Changes made. Jenkins job with passing result: 
https://mastern-jenkins-csb-openshift-qe.apps.ocp-c1.prod.psi.redhat.com/job/scale-ci/job/e2e-benchmarking-multibranch-pipeline/job/regression-test/40/console

File: perfscale_regression_ci/scripts/scalability/node-affinity-anti-affinity.sh [hunk 7]
```diff
@@ -0,0 +1,162 @@
...
+
+compute_nodes=$(oc get nodes -l 'node-role.kubernetes.io/worker=' | awk '{print $1}' | grep -v NAME | xargs)
+
+echo -e "\nWorker  nodes are: $compute_nodes"
+
+declare -a node_array
+counter=1
+
+oc get nodes -l 'node-role.kubernetes.io/worker='
+oc describe nodes -l 'node-role.kubernetes.io/worker='
```
Comment (paigerube14) [inline perfscale_regression_ci/scripts/scalability/node-affinity-anti-affinity.sh:47]: let's maybe move this to after the run workload is done. A check point of this test is to see how much cpu and memory we can push on a node, so this describe would be better seen after the wokrload is run before delete

File: perfscale_regression_ci/scripts/scalability/node-affinity-anti-affinity.sh [hunk 8]
```diff
@@ -0,0 +1,162 @@
...
+
+sleep 5
+
+
+echo "======Use kube-burner to load the cluster with test objects======"
+run_workload
+
+
+# Waiting for the pods in both namespaces - will add a check in the future
+sleep 60
```
[the remark this answers is not in the document]
Comment (paigerube14) [reply perfscale_regression_ci/scripts/scalability/node-affinity-anti-affinity.sh:91]: we should be able to use [POD_WAIT](https://github.com/cloud-bulldozer/e2e-benchmarking/tree/master/workloads/kube-burner) to do this instead. I think we will always want that functionality if you can add it to your kube burner config yaml file.

[the remark this answers is not in the document]
Comment (paigerube14) [reply perfscale_regression_ci/scripts/scalability/node-affinity-anti-affinity.sh:91]: the pod_wait will repoll the status' of all the pods instead of just waiting an arbitrary amount of time that the pods might or might not go running

[the remark this answers is not in the document]
Comment (paigerube14) [reply perfscale_regression_ci/scripts/scalability/node-affinity-anti-affinity.sh:91]: we can also use WAIT_FOR in the config file as well with an empty list (it'll equal all of the object types created)

**Comment by lenahorsley** [comment]
@paigerube14 @qiliRedHat 
Changes made. Jenkins job with passing result:
https://mastern-jenkins-csb-openshift-qe.apps.ocp-c1.prod.psi.redhat.com/job/scale-ci/job/e2e-benchmarking-multibranch-pipeline/job/regression-test/42/console

**Comment by paigerube14** [comment]
Looks good to me, I'll let Qiujie do a final review and merge
